src/button/default.scss

```diff
@@ -59,17 +53,34 @@
           linear-gradient(t(iui-color-background-1), t(iui-color-background-1));
       }
 
-      > .iui-label {
+      > .iui-button-label {
         @include themed {
           color: t(iui-color-foreground-primary);
         }
       }
+    }
 
-      > .iui-icon {
-        @include themed {
-          fill: t(iui-icons-color-primary);
-        }
-      }
+    > .iui-button-icon {
+      @include iui-button-icon(t(iui-icons-color-primary));
+    }
+  }
+
+  &[disabled],
+  &:disabled {
+    @include iui-button-disabled;
+  }
+}
+
+@mixin iui-button-default-base {
+  @include themed {
+    background-color: t(iui-color-background-1);
+    border-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-4));
+    color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
+  }
+
+  &.iui-active {
+    > .iui-button-icon {
+      @include iui-button-icon(t(iui-icons-color-primary));
```

Comment:
There are still two problems in this approach:
1. Including this mixin again will repeat all those rules
2. Plus all the nesting again makes it hard to override icon styles (i think this one is some 0-4-0)

I wonder if we should just change the value of fill CSS variable, instead of setting all those styles again. It won't work in IE but it will help

Reply:
Button not fully working in IE? Big no no no!

Reply:
it already doesn't fully work in both Safari & IE and no one complained 😆

Reply:
What? What is not working there?

Reply:
hover and focus states are broken in many of the button components (or derived ones like sidenav-button/header-button), because we use `&:hover, &:focus-visible`  in many places.

Reply:
I guess lets tackle CSS variables in another PR in 2022. Created issue #390 

And focus-visible should be fixed in this PR hopefully